website_portal_medical_prescription_order_line/controllers/main.py

```diff
@@ -39,17 +57,22 @@ def my_medical(self, **kw):
 
         pricelist = request.website.get_current_pricelist()
         pricelist_item_ids = pricelist.item_ids.ids
-        rx_lines_filtered = rx_line_ids.filtered(
-            lambda r: any(
-                i in r.medicament_id.item_ids.ids for i in pricelist_item_ids
-            )
-        )
 
-        all_patients = response.qcontext.get('patients', patient_model)
-        response.qcontext.update({
+        rx_lines_filtered = rx_obj.sudo().search([
```

Comment:
Is there a reason for doing a new search here instead of filtering the existing recordset and using ``sudo`` when that recordset is first generated at the top of the method? The latter would save a DB query.

Reply:
Nice. I did this because of an issue that ended up being unrelated. Went back to the filtering and added sudo to the original recs and it works just fine.